replica/table.cc

```diff
@@ -1501,8 +1501,8 @@ future<> table::clear() {
             _commitlog->discard_completed_segments(_schema->id(), t->get_and_discard_rp_set());
         }
     }
-    _memtables->clear_and_add();
-    return _cache.invalidate(row_cache::external_updater([] { /* There is no underlying mutation source */ }));
+    co_await _memtables->clear_and_add();
```

Comment:
Can this be called concurrently? Previously, clear_and_add() was atomic, but now two concurrent invocations can race.

Reply:
Yes, since `clear_and_add` now swaps the `_memtables` vector atomically and clears the old memtables in the background.

Reply:
Yes. Worth to add a comment so people don't add co_awaits before that exchange.